src/Monkeypatcher.cc

```diff
@@ -937,7 +937,22 @@ void patch_after_exec_arch<X64Arch>(RecordTask* t, Monkeypatcher& patcher) {
 }
 
 template <>
-void patch_after_exec_arch<ARM64Arch>(RecordTask*, Monkeypatcher&) {
+void patch_after_exec_arch<ARM64Arch>(RecordTask* t, Monkeypatcher& patcher) {
+  setup_preload_library_path<ARM64Arch>(t);
+  setup_audit_library_path<ARM64Arch>(t);
+
+  for (const auto& m : t->vm()->maps()) {
+    auto& km = m.map;
+    patcher.patch_after_mmap(t, km.start(), km.size(),
+                             km.file_offset_bytes()/page_size(), -1,
+                             Monkeypatcher::MMAP_EXEC);
+  }
+
+  if (!t->vm()->has_vdso()) {
+    patch_auxv_vdso(t);
```

Comment:
I'm confused, why is this in a !has_vdso branch?

Reply:
this patches out the vdso address from auxv, so that the application doesn't see it. Perhaps the function needs a better name.

Reply:
Lets file an issue on renaming it to `patch_auxv_to_remove_vdso` or something.